File: fixtures/csharp/controller-service-provider/src/Payments.Core/StripeProvider.cs
namespace Chakra.Payments.Provider;

public sealed class StripeProvider : IPaymentProvider
{
    private readonly string apiKey;

    public StripeProvider(string apiKey)
    {
        this.apiKey = apiKey;
    }

    public string ProviderLabel => "stripe";

    public async Task<string> RefundAsync(int amountCents)
    {
        await Task.Yield();
        return $"{apiKey}:{amountCents}";
    }
}

File: fixtures/java/controller-service-provider/src/main/java/chakra/payments/provider/StripeProvider.java
package chakra.payments.provider;

/** Concrete provider processing refunds against the Stripe API. */
public class StripeProvider implements PaymentProvider {

    private final String apiKey;

    public StripeProvider(String apiKey) {
        this.apiKey = apiKey;
    }

    @Override
    public boolean refund(long amountCents) {
        return amountCents > 0 && !this.apiKey.isEmpty();
    }
}
